### DISCUSSION TITLE
Index performance degrades over time

### DISCUSSION THREAD

**Discussion by bavaria95** [milvus-io/milvus#25178] [Q&A and General discussion]
Hi! We are still on Milvus 1.0 and noticed the following. At one point, if you search in a collection (1.8M vectors) by vector - it may find some items. But then in a week (might have added another 10-20k vectors), if you make the same query - it might not even find the same good matches it could find last week. And if taking all the vectors and doing brute force search - it obviously finds them in topK positions.
I understand that it's an approximate index, but I would still expect it to find the same items.

```python
# index params
param = {
        'dimension': 768,
        'index_file_size': 1024, 'metric_type': MetricType.IP,
}

# search params
{'nprobe': 16}
```

**Comment by yhmo** [comment]
Hmm... didn't hear this problem before. According to Milvus technical design, newly added data is stored in a new segment and doesn't affect previous segments(and previous index).
I remember an old bug is: milvus uses "metrictype.L2" for non-indexed segments and "metrictype.IP" for indexed segments. But this bug was fixed in 0.7 or 0.8.

**Reply by yhmo** [reply]
You can verify it by this way: re-index the collection by metrictype.L2. then query for one week(added another 10-20k vectors in the week).

**Reply by bavaria95** [reply]
I tried reindexing it with the same index type (as we use cosine similarity and not sure L2 would do here). It solved the issue for some time. But eventually recall degrades (the first time it was many many months, but we gradually were adding 1-3k items daily), now it happened after a week.
Also noticed that similarity scores aren't necessary the same. E.g. doing query search with vec1 shows similarity 0.65 with vec2, but if I extract vector and compute similarity (e.g. with sklearn) - it may show 0.69. Not sure it should be 100% accurate here though
